internal/util/testutil/equal.go

```diff
@@ -66,6 +66,12 @@ func AssertEqualSlices[T types.Type](tb testing.TB, expected, actual []T) bool {
 	return assert.Fail(tb, msg)
 }
 
+// EqualValue returns true when two BSON values are equal,
+// even if the numbers are different types.
```

Comment:
This methid looks very handy! I suggest to add a comment why `assert.EqualValues` from `testify` is not enough.

Reply:
No, we should not add more testing helpers that check that values are almost equal, but not quite. Instead, we should either fix our code or create separate issues and tests that will be skipped for now

Reply:
this method was removed in favour for not adding more test helpers